052-cruise-control-api-users.md

```diff
@@ -0,0 +1,220 @@
+* Experiment with Cruise Control operations not currently supported by Strimzi.
+* Debugging Cruise Control in a secured environments.
+
+At this time, Strimzi Cruise Control integration offers limited access to Cruise Control data and features.
+Although this limited access protects developers from preforming potentially dangerous cluster operations it also prevents developers from accessing harmless monitoring data.
+It also prevents developers from enabling third party applications such as the Cruise Control UI which could make monitoring their Cruise Control data easier.
+
+In order to allow developers and third-party applications to access the Cruise Control API without needing to disable API security, we would need to provide a method of creating API users.
+
+## Proposal
+
+### Goals
+
+* A method of creating Cruise Control API users.
+
+### Developer managed API user configuration
+
+Having the developer manage a single secret with their custom Cruise Control API user configuration would greatly reduce the complexity of the proposed design.
+Strimzi could add the contents of this secret to the Cruise Control API user configuration file containing Strimzi's `admin` and `healthcheck` API users.
+
+The schema could look something like this:
+
+```yaml=
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  ...
+spec:
+  cruiseControl:
+    apiUsersConfig:
+      secret:
+        secretName: cruise-control-api-users-secret
+     ...
+```
+
+Although this would simplify the overall design it would also shift a lot of the maintanence responsibility to the developer.
+The developer would need to create a valid Cruise Control API user configuration and encode it into a secret.
+That being said, this is still a reasonable amount of responsibility to leave to the developer.
+This is because accessing the Cruise Control API directly is neither the supported or primary way of interacting with Cruise Control for a Strimzi-managed Kafka cluster.
+
+### Risks
+
+Of course, supporting such a feature could come with some risks.
+If we were to support developers to create Cruise Control API users with the `ADMIN` role, those users would have access to potentially destructive cluster operations.
+For example, an API user with the ADMIN role would be able to execute a rebalance opeation while the Strimzi Operator is in the middle of preforming a rolling update of the Kafka brokers.
+
+That being said, supporting developers to create Cruise Control API users with non-`ADMIN` roles like `USER` and `VIEWER` roles would mitigate this risk completely.
+This is because the `USER` and `VIEWER` roles are read-only roles and would not conflict with Strimzi cluster operations.
+
+We could support the creation of API users with `ADMIN` roles with a caveat so that developers could experiment with Cruise Control write operations or we could simply disable it completely. 
```

Comment:
A proposal should be written in terms of what you _actually propose doing_. You can elaborate on why that's what's being proposed if necessary. And you can use the Rejected Alternatives section to forestall obvious variations which people might have.